Campaign/app/Http/Controllers/CampaignController.php

```diff
@@ -782,18 +791,9 @@ public function showtime(
             }
 
             // segment
-            foreach ($campaign->segments as $campaignSegment) {
-                $campaignSegment->setRelation('campaign', $campaign); // setting this manually to avoid DB query
-
-                if ($userId) {
-                    if (!$sa->checkUser($campaignSegment, strval($userId))) {
-                        continue 2;
-                    }
-                } else {
-                    if (!$sa->checkBrowser($campaignSegment, strval($browserId))) {
-                        continue 2;
-                    }
-                }
+            $canSee = $this->showtime->canSegmentSeeTheCampaign($campaign, $sa, $userId, $browserId);
```

Comment:
The business logic changed here. Before, user needed to be member of all segments to see the campaign. After this change, it would be enough if the user was member of one of the segments to see the campaign.

Unfortunately such change would break existing campaigns after update. To keep things consistent with existing implementation, we'll need to make a change to evaluate the segments this way:

* User needs to me member of all *inclusive* segments
* User must not be member of all *exclusive* segments

Only then, he/she will see the campaign.

Your campaigns now might expect the behavior you implemented - that it's enough if user belongs to one *inclusive* segment instead of all. In such case, you'll probably need to split your campaign into two and use *inclusive* each segment in one of the campaigns.

Reply:
Handled as described above, it'll get here once internal repo will get mirrored here.